src/lcm/grids.py

```diff
@@ -93,7 +95,7 @@ def to_jax(self) -> Array:
     def get_coordinate(self, value: Scalar) -> Scalar:
         """Get the generalized coordinate of a value in the grid."""
 
-    def replace(self, **kwargs: Any) -> "ContinuousGrid":  # noqa: ANN401
+    def replace(self, **kwargs: Any) -> ContinuousGrid:  # noqa: ANN401
```

Comment:
```suggestion
    def replace(self, **kwargs: int | float) -> ContinuousGrid:   
```

? (not sure whether that works, but we know the types)


Also, please just use `import dataclasses` above (same in `user_model.py`), we only use it once and I had to look up what was meant by `dc`.